### PR TITLE
New proxy realm provider config for 1.15

### PR DESCRIPTION
**Pull request by dLobatog**

### CODE REVIEW

File: _includes/manuals/1.15/4.3.8_smartproxy_realm.md [hunk 1]
```diff
@@ -6,11 +6,13 @@ Activate the realm management module within the Smart Proxy instance.  This mana
 :realm_provider: freeipa
```
Comment (domcleal) [inline _includes/manuals/1.15/4.3.8_smartproxy_realm.md]: 1. `:use_provider`
1. `realm_freeipa`

File: _includes/manuals/1.15/4.3.8_smartproxy_realm.md [hunk 2]
```diff
@@ -6,11 +6,13 @@ Activate the realm management module within the Smart Proxy instance.  This mana
...
 </pre>
 
-`freeipa` is the only provider currently available.
+We have a provider for `freeipa` available. The configuration for each provider should be
```
Comment (domcleal) [inline _includes/manuals/1.15/4.3.8_smartproxy_realm.md]: 1. Avoid pronouns - perhaps use the same format (a bullet point list) as other sections (DNS, DHCP) with provider support.
1. It's the `realm_freeipa` provider, or it's _for FreeIPA_, or both. This should probably read: "`realm_freeipa` - host object management in FreeIPA"

File: _includes/manuals/1.15/4.3.8_smartproxy_realm.md [hunk 3]
```diff
@@ -6,11 +6,13 @@ Activate the realm management module within the Smart Proxy instance.  This mana
...
+in its respective file, i.e: `/etc/foreman-proxy/settings.d/realm_freeipa.yml`.
 
-The following settings control authentication of the proxy to the realm for management of hosts:
+The following settings control authentication of the proxy to the realm for management of hosts.
+On `realm_freeipa.yml`:
 <pre>
 # Authentication for Kerberos-based Realms
-:realm_keytab: /etc/foreman-proxy/freeipa.keytab
-:realm_principal: realm-proxy@EXAMPLE.COM
+:keytab: /etc/foreman-proxy/freeipa.keytab
```
Comment (domcleal) [inline _includes/manuals/1.15/4.3.8_smartproxy_realm.md]: `keytab_path`

File: _includes/manuals/1.15/4.3.8.2_freeipa_realm.md [hunk 4]
```diff
@@ -32,7 +32,7 @@ the correct permissions:
         chown foreman-proxy /etc/foreman-proxy/freeipa.keytab
         chmod 600 /etc/foreman-proxy/freeipa.keytab
 
-Then update realm.yml with the relevant settings.
+Then update `settings.d/realm_freeipa.yml` with the relevant settings.
 
 If you're using FreeIPA to manage DNS records, and want them to be automatically deleted when the host is deleted in Foreman, set this to true:
```
Comment (domcleal) [inline _includes/manuals/1.15/4.3.8.2_freeipa_realm.md:37]: Line below this should be merely `:remove_dns: true`.

File: _includes/manuals/1.15/4.3.8_smartproxy_realm.md [hunk 5]
```diff
@@ -6,11 +6,13 @@ Activate the realm management module within the Smart Proxy instance.  This mana
 :realm_provider: freeipa
 </pre>
 
-`freeipa` is the only provider currently available.
+We have a provider for `freeipa` available. The configuration for each provider should be
+in its respective file, i.e: `/etc/foreman-proxy/settings.d/realm_freeipa.yml`.
 
-The following settings control authentication of the proxy to the realm for management of hosts:
+The following settings control authentication of the proxy to the realm for management of hosts.
+On `realm_freeipa.yml`:
```
Comment (domcleal) [inline _includes/manuals/1.15/4.3.8_smartproxy_realm.md]: s/on/in/

**Review by domcleal — CHANGES_REQUESTED** [review]

**Comment by dLobatog** [comment]
Updated, thanks @domcleal

File: _includes/manuals/1.15/4.3.8_smartproxy_realm.md [hunk 6]
```diff
@@ -3,14 +3,19 @@ Activate the realm management module within the Smart Proxy instance.  This mana
...
 :enabled: https
-:realm_provider: freeipa
+:use_provider: realm_freeipa
 </pre>
 
-`freeipa` is the only provider currently available.
+Builtin providers are:
 
-The following settings control authentication of the proxy to the realm for management of hosts:
+* 'realm_freeipa' -  host object management in FreeIPA
```
Comment (domcleal) [inline _includes/manuals/1.15/4.3.8_smartproxy_realm.md]: Use backticks for code formatting.

**Review by domcleal — CHANGES_REQUESTED** [review]

**Comment by dLobatog** [comment]
Updated now with backticks on `realm_freeipa`

**Review by domcleal — APPROVED** [review]

**Comment by domcleal** [comment]
Thanks @dLobatog.

**Pull request merged**